charts/consul/templates/server-acl-init-job.yaml

```diff
@@ -33,12 +33,23 @@ spec:
         component: server-acl-init
       annotations:
         "consul.hashicorp.com/connect-inject": "false"
+        {{- if (and .Values.global.secretsBackend.vault.enabled .Values.global.tls.enabled) }}
+        "vault.hashicorp.com/agent-pre-populate-only": "true"
```

Comment:
Not your fault but I'm not sure why this is necessary ([vault api docs aren't super clear](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-agent-pre-populate)), what is this super secret annotation used for?

Reply:
This is currently the only way to attach the vault agent as a sidecar and reliably have the job complete with the sidecar.